chain/actors/builtin/builtin.go

```diff
@@ -41,3 +29,23 @@ func FromV0FilterEstimate(v0 smoothing0.FilterEstimate) FilterEstimate {
 func QAPowerForWeight(size abi.SectorSize, duration abi.ChainEpoch, dealWeight, verifiedWeight abi.DealWeight) abi.StoragePower {
 	return miner0.QAPowerForWeight(size, duration, dealWeight, verifiedWeight)
 }
+
+func FromV1FilterEstimate(v1 smoothing1.FilterEstimate) FilterEstimate {
+	return (FilterEstimate)(v1)
+}
+
+type ActorStateLoader func(store adt.Store, root cid.Cid) (cbor.Marshaler, error)
+
+var ActorStateLoaders = make(map[cid.Cid]ActorStateLoader)
+
+func RegisterActorState(code cid.Cid, loader ActorStateLoader) {
+	ActorStateLoaders[code] = loader
+}
+
+func Load(store adt.Store, act *types.Actor) (cbor.Marshaler, error) {
+	loader, found := ActorStateLoaders[act.Code]
+	if !found {
+		return nil, xerrors.Errorf("unknown actor code %s", act.Code)
+	}
+	return loader(store, act.Head)
+}
```

Comment:
This lets us "load" an arbitrary actor. We can then type switch on it, or use reflection.

I'm not really sure if this is worth it. It lets us implement `vm.MutateState` but it introduces a lot of code and it's not hard to just implement `vm.MutateActor` and call `actorType.Load`.

Thoughts? Can you think of other places where this would be useful>